monkeytype/stubs.py

```diff
@@ -298,31 +299,64 @@ def _get_optional_elem(anno: Any) -> Any:
     return Union[elems]
 
 
-def render_annotation(anno: Any) -> str:
-    """Convert an annotation into its stub representation."""
-    if _is_optional(anno):
-        elem_type = _get_optional_elem(anno)
-        rendered = 'Optional[' + render_annotation(elem_type) + ']'
-    elif hasattr(anno, '__supertype__'):
-        rendered = anno.__name__
-    elif is_forward_ref(anno):
-        rendered = repr(anno.__forward_arg__)
-    elif getattr(anno, '__module__', None) == 'typing':
-        rendered = repr(anno).replace('typing.', '')
-    elif isinstance(anno, NoneType):
-        rendered = 'None'
-    elif isinstance(anno, type):
-        if anno.__module__ in ('builtins',):
-            rendered = anno.__qualname__
+class RenderAnnotation(GenericTypeRewriter[str]):
+    """Render annotation recursively."""
```

Comment:
This is awesome! You just fixed another significant longstanding bug in passing :)